File: instrumentation/common-api/src/main/java/io/opentelemetry/android/instrumentation/common/DefaultingActivityLifecycleCallbacks.java
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.android.instrumentation.common;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Interface helper for implementations that don't need/want all the extra baggage of the full
 * Application.ActivityLifecycleCallbacks interface. Implementations can choose which methods to
 * implement.
 */
public interface DefaultingActivityLifecycleCallbacks
        extends Application.ActivityLifecycleCallbacks {

    @Override
    default void onActivityCreated(
            @NonNull Activity activity, @Nullable Bundle savedInstanceState) {}

    @Override
    default void onActivityStarted(@NonNull Activity activity) {}

    @Override
    default void onActivityResumed(@NonNull Activity activity) {}

    @Override
    default void onActivityPaused(@NonNull Activity activity) {}

    @Override
    default void onActivityStopped(@NonNull Activity activity) {}

    @Override
    default void onActivitySaveInstanceState(
            @NonNull Activity activity, @NonNull Bundle outState) {}

    @Override
    default void onActivityDestroyed(@NonNull Activity activity) {}
}

File: core/src/main/java/io/opentelemetry/android/internal/services/visiblescreen/activities/DefaultingActivityLifecycleCallbacks.kt
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.android.internal.services.visiblescreen.activities

import android.app.Activity
import android.app.Application.ActivityLifecycleCallbacks
import android.os.Bundle

/**
 * Interface helper for implementations that don't need/want all the extra baggage of the full
 * Application.ActivityLifecycleCallbacks interface. Implementations can choose which methods to
 * implement.
 */
interface DefaultingActivityLifecycleCallbacks : ActivityLifecycleCallbacks {
    override fun onActivityCreated(
        activity: Activity,
        savedInstanceState: Bundle?,
    ) { }

    override fun onActivityStarted(activity: Activity) {}

    override fun onActivityResumed(activity: Activity) {}

    override fun onActivityPaused(activity: Activity) {}

    override fun onActivityStopped(activity: Activity) {}

    override fun onActivitySaveInstanceState(
        activity: Activity,
        outState: Bundle,
    ) {
    }

    override fun onActivityDestroyed(activity: Activity) {}
}
